doc/transactions.md

```diff
@@ -0,0 +1,91 @@
+Transactions
+============
+Transactions is a feature that comes from SOFT, which allows to easy manage arbitrary long series of immutable (frozen) instances, while ensuring provenance.
+Conceptually, it shares many similarities with [git](https://git-scm.com).
+
+A basic usage for transactions is to take snapshots of the current state of your system.
+This is shown in Figure 1, where we assume that the instance `A` (which, e.g., could be a collection) describes the state of your system.
+
+We call an instance a *transaction* after we have taken a snapshot of it, i.e. when it contains a reference to an immutable previous ("parent") instance.
```

Comment:
Is it possible to take a snapshot that is exactly similar in content to a previous one? This should be mentioned somewhere in this section.

In the figure example, e.g., it is unclear whether the _values_/content of `A1` has been changed, or the time `t1` is merely a meta-data value.
In general, the concept of time in these examples are a bit tenuous at best. It should be made more explicit - either by setting up a thought example about a time-resolved experiment, where time is a value and property of the underlying data model, or in another way that makes it clear what `t1` and `t2` are, other than abstract moments in time.

Reply:
Just to be clear about nomenclature. When dealing with transactions we have to separate between identity, state and values. A value cannot change, 3 will always be 3, but the age of an item can change over time, but at any given time have a specific state. The state at a given point in time (specific instance of state/snapshot) is a value, and cannot be changed. identity is related to a sequence of states (snapshots/transactions) and point to different values (immutable) at a specific point in time (state).